### PR TITLE
Giao dien user structure

### PR DESCRIPTION
**Pull request by NgengChhengkim**

### CODE REVIEW

File: app/datatables/locations_datatable.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def data
       manager = location.manager
       [
         index + 1,
-        link_to(location.name, "#"),
+        link_to(location.name, @view.admin_location_path(location)),
```
Comment (nguyenngocha) [inline app/datatables/locations_datatable.rb]: thu bo () quanh location xem duoc khong a.

File: app/models/user.rb [hunk 2]
```diff
@@ -55,10 +55,13 @@ class User < ApplicationRecord
   scope :show_members, ->{order(:role_id, :name).limit Settings.number_member_show}
   scope :select_all, ->{joins :role}
   scope :not_trainees, ->{joins(:role).where("roles.name != 'trainee'")}
+  scope :by_location, ->location_id{
+    joins(:profile).where("profiles.location_id = ?", location_id)
```
Comment (nguyenngocha) [inline app/models/user.rb]: location_id: location_id

File: app/views/admin/users/_form.html.erb [hunk 3]
```diff
@@ -13,7 +13,10 @@
           <%= f.label :role_id %>
           <%= f.select :role_id, options_for_select(@roles.collect{|role| [role.name,
             role.id]}, f.object.role_id), {}, class: "form-control" %>
-        <% end %><br>
+        <% end %>
+
+        <%= f.label :trainer_id %>
+        <%= f.select :trainer_id, options_for_select(@trainers.collect{|trainer| [trainer.name, trainer.id]}, f.object.trainer_id), {include_blank: true}, class: "form-control" %>
```
Comment (nguyenngocha) [inline app/views/admin/users/_form.html.erb]: dai qua ...

File: app/datatables/locations_datatable.rb (hunk 1 above)
Comment (NgengChhengkim) [inline app/datatables/locations_datatable.rb]: bo khong dc

File: app/models/user.rb (hunk 2 above)
Comment (NgengChhengkim) [inline app/models/user.rb]: khong chay duoc

**Pull request merged**